### PR TITLE
feat: index top 50 pypi packages

### PR DESCRIPTION
**Pull request by zsol**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92
* __->__ #89
* #88
* #90

### CODE REVIEW

**Comment by abesto** [comment]
This diff looks to me like it's using the new async goodness to actually do work faster. It doesn't... find or contain a list of top 50 pypi packages? Confuse.

File: py_wtf/cli/__init__.py [hunk 1]
```diff
@@ -64,6 +70,42 @@ async def index(project_name: str, directory: str, pretty: bool, force: bool) ->
...
+    out_dir = Path(directory)
+    out_dir.mkdir(parents=True, exist_ok=True)
+    repo = ProjectRepository(out_dir)
+    async with httpx.AsyncClient() as client:
+        top_pkgs = (
+            await client.get(
+                "https://hugovk.github.io/top-pypi-packages/top-pypi-packages-30-days.json"
+            )
+        ).json()
+        projects: Iterable[str] = (row["project"] for row in top_pkgs["rows"][:top])
```
Comment (zsol) [inline py_wtf/cli/__init__.py:87]: @abesto lookie here

**Comment by zsol** [comment]
I can break it up into moar PRs if that makes it easier to review, lmk

File: py_wtf/cli/__init__.py (hunk 1 above)
Comment (abesto) [reply py_wtf/cli/__init__.py:87]: Ah! How... did I miss that. Oh well.

Comment (abesto) [reply py_wtf/cli/__init__.py:87]: ... swear this looked different when I last checked.

File: py_wtf/cli/__init__.py [hunk 2]
```diff
@@ -64,6 +70,43 @@ async def index(project_name: str, directory: str, pretty: bool, force: bool) ->
...
+
+    with rich.progress.Progress(
+        rich.progress.TimeElapsedColumn(),
+        rich.progress.TextColumn("{task.fields[action]} {task.description}"),
+        rich.progress.BarColumn(),
```
Comment (abesto) [inline py_wtf/cli/__init__.py:92]: Noice

**Review by abesto — APPROVED** [review]

**Pull request merged**